File: 12-BIT-MANIPULATION/PHP/reverse-bits.php
<?php

// https://leetcode.com/problems/reverse-bits/

class Solution
{
    function reverseBits(int $n): int
    {
        $res = 0;
        for ($i = 0; $i < 32; $i++) {
            $bit = ($n >> $i) & 1;
            $res |= $bit << (31 - $i);

        }

        return $res;
    }
}

File: 12-BIT-MANIPULATION/Go/reverse-bits.go
package main

// https://leetcode.com/problems/reverse-bits/

func reverseBits(num uint32) uint32 {
    var res uint32
    for i := 0; i < 32; i++ {
        bit := (num >> i) & 1
        res |= bit << (31 - i)
    }

    return res
}